include/circt/Dialect/Verif/VerifOps.td

```diff
@@ -260,4 +262,91 @@ def YieldOp : VerifOp<"yield", [
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// Verification intent ops
+//===----------------------------------------------------------------------===//
+
+/// BMC -- short for bounded model check.
+def BMC : I32EnumAttrCase<"BMC", 0, "bmc">;
+/// KIP -- short for k-Inductive-Proof.
+def KInductiveProof : I32EnumAttrCase<"KInductiveProof", 1, "kinduction">;
+
+def FormalTestTypeAttr : I32EnumAttr<"FormalTestType", "formal test type",
+                                [BMC, KInductiveProof]> {
+  let cppNamespace = "circt::verif";
+}
+
+def FormalOp : VerifOp<"formal", [NoTerminator, RegionKindInterface]> {
+  let summary = "defines a formal verification test";
+  let description = [{
+    This operation defines a formal verification test with potentially additional
+    assertions and assumptions. This should be written by declaring symbolic 
```

Comment:
`potentially additional assertions and assumptions` might be a bit confusing. Are they any different from regular assertions and assumptions in the design? AFAIU you could arbitrarily pull out assertions from the top level module or pull them in. You could also just inline the whole design into the formal op, right?